rmf_site_editor/src/site/save.rs

```diff
@@ -209,6 +211,17 @@ fn assign_site_ids(world: &mut World, site: Entity) -> Result<(), SiteGeneration
         }
 
         if let Ok(lift) = lifts.get(*site_child) {
+            if let Ok(anchor_group) = cabin_anchor_groups.get(*site_child) {
+                if let Ok(anchor_children) = children.get(**anchor_group) {
+                    for anchor_child in anchor_children {
+                        if let Ok(e) = level_children.get(*anchor_child) {
```

Comment:
This is not the best naming, I reused the `level_children` query that already queries for anchors, but I can create a duplicated `anchors` query if we want the cleanliness

Reply:
`level_children` queries for a lot more things than just anchors, so I definitely think we should not be using it for this. As an example of why.. It's plausible that at some point we'd implement animated lift doors by recycling the `DoorType` component inside a child of the lift. However we don't necessarily want that child entity to have its own SiteID since it's really part of the same "site element" as the lift. If we assign SiteIDs to all these different child types without considering whether it's really an element that gets saved, then we'll be contaminating the SiteIDs.

Reply:
Looking at the previously existing implementation, I'm realizing that we were already assigning SiteIDs to `level_children` entities within lifts. Now I'm conflicted and wondering if there was a reason we were doing this previously, e.g. to allow walls and models to be placed inside of lifts.